test/Microsoft.Azure.WebJobs.Extensions.EventHubs.Tests/EventHubApplicationInsightsTest.cs

```diff
@@ -0,0 +1,438 @@
+            {
+                await host.StartAsync();
+
+                var method = typeof(EventHubTestMultipleDispatchJobs).GetMethod("SendEvents_TestHub", BindingFlags.Static | BindingFlags.Public);
+                await host.GetJobHost().CallAsync(method, new { numEvents = 5, input = _testPrefix });
+
+                bool result = _eventWait.WaitOne(Timeout);
+                Assert.True(result);
+
+                await host.StopAsync();
+            }
+
+            List<RequestTelemetry> requests = _channel.Telemetries.OfType<RequestTelemetry>().ToList();
+            List<DependencyTelemetry> dependencies = _channel.Telemetries.OfType<DependencyTelemetry>().ToList();
+
+            // One dependency for the 'Send' from SendEvent_TestHub
+            Assert.Single(dependencies);
+            Assert.Single(dependencies, d => d.Name == "Send");
+
+            var ehOutDependency = dependencies.Single(d => d.Name == "Send");
+
+            var manualCallRequest = requests.Single(r => r.Context.Operation.Name == "SendEvents_TestHub");
+            var manualOperationId = manualCallRequest.Context.Operation.Id;
+
+            ValidateEventHubDependency(
+                ehOutDependency,
+                _endpoint,
+                TestHubName,
+                "Send",
+                "SendEvents_TestHub",
+                manualOperationId,
+                manualCallRequest.Id,
+                LogCategories.Bindings);
+
+            var ehTriggerRequests = requests.Where(r => r.Context.Operation.Name == "ProcessMultipleEvents");
+            List<TestLink> actualLinks = new List<TestLink>();
+            foreach (var ehTriggerRequest in ehTriggerRequests)
+            {
+                if (ehTriggerRequest.Properties.TryGetValue("_MS.links", out var linksStr))
+                {
+                    ValidateEventHubRequest(
+                        ehTriggerRequest,
+                        true,
+                        "ProcessMultipleEvents",
+                        null,
+                        null);
+
+                    Assert.NotNull(ehTriggerRequest.Context.Operation.Id);
+                    Assert.Null(ehTriggerRequest.Context.Operation.ParentId);
+
+                    actualLinks.AddRange(JsonConvert.DeserializeObject<TestLink[]>(linksStr, jsonSettingThrowOnError));
+                }
+            }
+
+            Assert.True(actualLinks.Count >= 5); // we've sent 5 events
+
+            var currentTestLinks = actualLinks.Where(l => l.operation_Id == manualOperationId).ToArray();
+
+            // there should be 5 relevant links
+            // current Event Hubs SDK does not generate unique Id per message, 
+            // so they all share the same Id
+            Assert.Equal(5, currentTestLinks.Length); 
+            Assert.Equal(5, currentTestLinks.Count(l => l.id == ehOutDependency.Id));
+        }
+
+        [Fact(Skip = "Requires Microsoft.Azure.WebJobs.Logging.ApplicationInsights v 3.0.13+")]
```

Comment:
Were you just waiting for this to be published? I think this is live now.

Reply:
nope, there is more work to do (some PRs that not merged, and not even created yet).  So I'll change version to something else (14+)